sdk/core/azure-core/src/http/http.cpp

```diff
@@ -25,153 +28,30 @@ const HttpMethod HttpMethod::Put("PUT");
 const HttpMethod HttpMethod::Delete("DELETE");
 const HttpMethod HttpMethod::Patch("PATCH");
 
+namespace {
+bool IsInvalidHeaderNameChar(char c)
```

Comment:
Wouldn't this be slower than a single lookup in the `validChars` array, we had before?

Reply:
It is likely slightly slower, but I think it is way easier to support (less lines) and therefor has a higher chance to have less bugs. It is unlikely to be a perf bottleneck, especially in the SDK that talks to the network, and I'd say until we get data that it is a perf bottleneck, we should prefer ~5 line version to 256+ lines version. The version we had before did rely on comments, on 0 not being in the wrong spot, took at least 256 bytes, and it would have been harder to ensure it works correctly by reading it, than reading an isalnum and ~10 extra allowed character that fit on the screen and comparison is straighforward and not index-based.